web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -0,0 +1,204 @@
+                        keyColumnTypes.toArray(new String[0]),
+                        key.map((p0, p1, p2) -> JsArray.of((Object) p0).asArray(new Object[0]))
+                                .asArray(new Object[0][]),
+                        null,
+                        this).then(table -> {
+                            GetTableRequest getTableRequest = new GetTableRequest();
+                            getTableRequest.setPartitionedTable(widget.getTicket());
+                            getTableRequest.setKeyTableTicket(table.getHandle().makeTicket());
+                            getTableRequest.setResultId(cts.getHandle().makeTicket());
+                            connection.partitionedTableServiceClient().getTable(getTableRequest, connection.metadata(),
+                                    c::apply);
+                            return null;
+                        });
+            },
+                    "tablemap key " + key);
+
+            // later, when the CTS is released, remove this "table" from the map and replace with an unresolved JsLazy
+            entry.onRunning(
+                    ignore -> {
+                    },
+                    ignore -> {
+                    },
+                    () -> populateLazyTable(key, index));
+
+            // we'll make a table to return later, this func here just produces the JsLazy of the CTS
+            return entry.refetch(this, connection.metadata());
+
+        }));
+    }
+
+
+    public Promise<JsTable> getTable(Object key) {
+        // Wrap non-arrays in an array so we are consistent with how we track keys
+        if (!JsArray.isArray(key)) {
+            key = JsArray.of(key);
+        }
+        // Every caller gets a fresh table instance, and when all are closed, the CTS will be released.
+        // See #populateLazyTable for how that is tracked.
+        // noinspection SuspiciousMethodCalls - we know it is an array, see above
+        final JsLazy<Promise<ClientTableState>> entry = tables.get(key);
+        if (entry == null) {
+            // key doesn't even exist, just hand back a null table
+            return Promise.resolve((JsTable) null);
+        }
+        return entry.get().then(cts -> Promise.resolve(new JsTable(cts.getConnection(), cts)));
+    }
+
+    public Promise<JsTable> getMergedTable() {
+        return connection.newState((c, cts, metadata) -> {
+            MergeRequest requestMessage = new MergeRequest();
+            requestMessage.setPartitionedTable(widget.getTicket());
+            requestMessage.setResultId(cts.getHandle().makeTicket());
+            connection.partitionedTableServiceClient().merge(requestMessage, connection.metadata(), c::apply);
+        }, "tablemap merged table")
+                .refetch(this, connection.metadata())
+                .then(cts -> Promise.resolve(new JsTable(cts.getConnection(), cts)));
+    }
+
+    public JsSet<Object> getKeys() {
+        if (keys.getColumns().length == 1) {
```

Comment:
We wrap them internally, but the TableMap api has historically allowed passing in a single string as distinct from a string in an array. For example, in DHE, these two statements do different things:

```
const t = await tablemap.getTable("i'm a string")
```
```
const t = await tablemap.getTable(["i'm an array of strings"])
```
and there are cases where one vs the other is correct and appropriate.

For this method, I was seeking to return a plain string if the partitioned table was only partitioned on a single (string) column, so that .getKeys() returns an set of strings in that case, rather than an set of array of strings. Note that in this version of the code, the API can tell the difference and decide what to do - in DHE we cannot tell the difference, since it is possible to have heterogeneous key types.